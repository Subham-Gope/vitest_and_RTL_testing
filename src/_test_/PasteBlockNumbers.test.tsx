// import { describe, test, expect } from "vitest";
// import { screen, render } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import PasteBlockNumbers from "../components/PasteBlockNumbers";

// describe("Testing the PasteBlockNumbers.tsx file", () => {
//   test("pastes text to clipboard after checking the input is non-number", async () => {
//     // Setting up the userEvent
//     const user = userEvent.setup();

//     // Setting up the debugger
//     screen.debug();

//     // Setting snapshot matching
//     const { container } = render(<PasteBlockNumbers />);
//     expect(container).toMatchSnapshot();

//     const input = screen.getByRole("textbox", {
//       name: /paste you text here./i,
//     });
//     expect(input).toBeInTheDocument();

//     // Initiating user interactions

//     const text = "Hello there...";
//     input.focus();
//     await user.paste(text);

//     expect(input).toHaveValue(text);

//     const displayText = screen.getByText(/pasted text:/i);
//     expect(displayText).toBeInTheDocument();
//   });
// });
