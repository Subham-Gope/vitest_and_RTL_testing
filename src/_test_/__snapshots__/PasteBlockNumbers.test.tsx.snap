// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the PasteBlockNumbers.tsx file > pastes text to clipboard after checking the input is non-number 1`] = `
<div>
  <div>
    <label
      for="paste"
    >
      Paste you text here.
    </label>
    <input
      id="paste"
      placeholder="paste here"
      type="text"
      value=""
    />
  </div>
</div>
`;
